.github/workflows/perf.yml

```diff
@@ -0,0 +1,122 @@
+name: libp2p perf test
+
+# How to configure a repository for running this workflow:
+# 1. Run 'make ssh-keygen' in 'perf' to generate a new SSH key pair named 'user' in 'perf/terraform/region/files'
+# 2. Export AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY for the account of your choice
+# 3. Run 'terraform apply' in 'perf/terraform' to create the AWS resources
+# 4. Run 'terraform output' in 'perf/terraform' to get the bucket name
+# 5. Go to https://console.aws.amazon.com/iamv2/home?#/users/details/perf?section=security_credentials
+# 6. Click 'Create access key' to get the access key ID and secret access key
+# 7. Go to https://github.com/libp2p/test-plans/settings/secrets/actions
+# 8. Click 'New repository secret', set the name to 'PERF_AWS_SECRET_ACCESS_KEY', and paste the secret access key from step 6
+# 9. Click 'New repository secret', set the name to 'PERF_SSH_PRIVATE_KEY', and paste the private key from step 1
+# 10. Go to https://github.com/libp2p/test-plans/settings/variables/actions
+# 11. Click 'New repository variable', set the name to 'PERF_AWS_ACCESS_KEY_ID', and paste the access key ID from step 6
+# 12. Click 'New repository variable', set the name to 'PERF_AWS_BUCKET', and paste the bucket name from step 4
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    # Example:
+    # uses: libp2p/test-plans/.github/workflows/perf.yml@master
+    # with:
+    #   aws-access-key-id: ${{ vars.PERF_AWS_ACCESS_KEY_ID }}
+    #   aws-bucket: ${{ vars.PERF_AWS_BUCKET }}
+    #   ref: master
+    # secrets:
+    #   PERF_AWS_SECRET_ACCESS_KEY: ${{ secrets.PERF_AWS_SECRET_ACCESS_KEY }}
+    #   PERF_SSH_PRIVATE_KEY: ${{ secrets.PERF_SSH_PRIVATE_KEY }}
+    inputs:
+      aws-access-key-id:
+        type: string
+        required: true
+        description: The AWS access key ID of 'aws_iam_user.perf'
+      aws-bucket:
+        type: string
+        required: true
+        description: The AWS bucket as output by 'output.bucket_name'
+      ref:
+        type: string
+        required: false
+        description: The ref of the test-plans repo to use (defaults to 'master')
+        default: master
+    secrets:
+      PERF_AWS_SECRET_ACCESS_KEY: # The AWS secret access key of 'aws_iam_user.perf'
+        required: true
+      PERF_SSH_PRIVATE_KEY: # The SSH private key for 'aws_key_pair.perf'
+        required: true
+
+jobs:
+  perf:
+    name: Perf
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+    defaults:
+      run:
+        shell: bash
+        working-directory: perf
+    env:
+      AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id || vars.PERF_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.PERF_AWS_SECRET_ACCESS_KEY }}
+    steps:
+      - name: Configure SSH
+        uses: webfactory/ssh-agent@d4b9b8ff72958532804b70bbe600ad43b36d5f2e # v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.PERF_SSH_PRIVATE_KEY }}
```

Comment:
I guess with the move to *AWS launch templates* there is no easy way for ephemeral SSH keys? Ephemeral keys would eliminate the need to manage long lived credentials. Please ignore in case ephemeral keys would add more complexity.

Reply:
We could keep generating keys on the fly. We still do have to keep managing the long-lived AWS credentials. And if SSH keys are ephemeral, then those AWS creds have to be allowed to create AWS key pairs.

I'll add instructions on what it'd take to switch between the two next to the `aws_key_pair` resource and we can decide then. I don't feel strongly either way.